opsdroid/constraints.py

```diff
@@ -31,7 +31,12 @@ def constraint_decorator(func):
 
         def constraint_callback(message, rooms=rooms):
             """Check if the room is correct."""
-            return message.target in lookup_target(message, rooms)
+            if hasattr(message.connector, "lookup_target"):
+                return message.target in list(
+                    map(message.connector.lookup_target, rooms)
+                )
```

Comment:
If you do this here, then it will be evaluated below
```suggestion
                rooms = list(map(message.connector.lookup_target, rooms))
```

Reply:
Yea, what a silly mistake...

Reply:
No worries, It still worked, only a marginal improvement!